configmgr/audio_config.c

```diff
@@ -2287,7 +2291,13 @@ static int parse_stream_start(struct parse_state *state)
         return -EINVAL;
     }
 
-    if (attrib_to_uint(&card, state, e_attrib_card) == -EINVAL) {
+    if (attrib_to_uint(&card, state, e_attrib_card) == 0) {
+        if (state->attribs.value[e_attrib_cardname] != NULL) {
+            ALOGE("stream must be configured by only one of 'card' OR 'cardname'. Both provided.");
+            return -EINVAL;
+        }
+    } else if (get_card_id_for_name(state->attribs.value[e_attrib_cardname],
+                                    &card) != 0) {
```

Comment:
This breaks the parser. A 'stream' element normally doesn't specify a card attribute. But this code change will fail parsing if it doesn't have one of 'card' or 'cardname' attribute.

Reply:
Fixed.